### PR TITLE
[W.I.P]Adds ShowModule command and adds ModuleDisplay UI, fixes bug

### LINKED ISSUE #196: Add UI for display of module details
The details of module card should be shown inside browser panel when the user clicks on the module card.

### LINKED ISSUE #246: [PE Dry Run]Duplicate Modules for Staged and Taken
I can add the same module for staged and taken; should I be able to have the same module in both lists simultaneously?

<hr>

**Reported by:** @josephambe
**Severity:** `Low`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#757]</sub>

### LINKED ISSUE #265: Create ShowModule command to show the module info by CLI

### PR DESCRIPTION
**Pull request by hzf1174**
This PR is for the summary of milestone mid v1.4. In this PR, a new feature called `showModule` has been added to display a certain module in the database. The UI for `ModuleDisplay` and can be shown when user clicks on the `ModuleCard` or use `ShowModule` command. Bug mentioned in PE dry run has been fixed. 

Logic
ShowModule command has been added. It uses a `Code` as key to search a certain module in the database and show the module information to user. It is equivalent to click on that module card. `Add` 
commands now also check whether another module list(staged/taken) has duplicate module as it does not make sense for two lists have the same module. `List` command now is available.

Model
Model#searchForIndexInDatabase(...) has been added to search the `Index` of a certain module. It is used for ShowModule command. ModuleList and UniqueModuleList are updated respectively.

UI
ModuleDisplay has been implemented. Except `prereq`, all the information of a module has been displayed. ModuleListPanel and its related files/methods have been removed. `PanelSelectionChangedEvent` now is handled by `MainWindow` instead of `BrowserPanel`.

Commons
JumpToDatabaseListRequestEvent has been added. It is used for showModule method.

Testing
New test cases for add/showModule command  and displayModule UI have been added.

The following issues have been resolved:
Resolves #196 
Resolves #246 
Resolves #265

### CODE REVIEW

File: src/main/java/seedu/modsuni/commons/events/ui/JumpToDatabaseListRequestEvent.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package seedu.modsuni.commons.events.ui;
+
+import seedu.modsuni.commons.core.index.Index;
+import seedu.modsuni.commons.events.BaseEvent;
+
+/**
+ * Indicates a request to jump to the list of persons
```
Comment (RTJL) [inline src/main/java/seedu/modsuni/commons/events/ui/JumpToDatabaseListRequestEvent.java]: Is it supposed to be list of persons?

File: src/test/java/seedu/modsuni/testutil/TypicalModules.java [hunk 2]
```diff
@@ -56,6 +56,21 @@
...
+                    + "development, namely object-oriented system analysis, object-oriented system modelling and "
+                    + "design, implementation, and testing, with emphasis on system modelling and design and "
+                    + "implementation of software modules that work cooperatively to fulfill the requirements of "
+                    + "the system. Tools and techniques for software development, such as Unified Modelling "
+                    + "Language (UML), program specification, and testing methods, will be taught. Major software "
+                    + "engineering issues such as modularisation criteria, program correctness, "
+                    + "and software quality will also be covered.")
+            .withCredit(4).withIsAvailableInSem1(true).withIsAvailableInSem2(true)
+            .withisAvailableInSpecialTerm1(false).withisAvailableInSpecialTerm2(false)
+            .withLockedModules(new ArrayList<Code>()).withPrereq(new Prereq()).build();
```
Comment (RTJL) [inline src/test/java/seedu/modsuni/testutil/TypicalModules.java]: Would be good to add in the actual prerequisites & locked modules.

Comment (RTJL) [reply src/test/java/seedu/modsuni/testutil/TypicalModules.java]: Not critical at the moment.

File: src/main/java/seedu/modsuni/commons/events/ui/JumpToDatabaseListRequestEvent.java (hunk 1 above)
Comment (hzf1174) [reply src/main/java/seedu/modsuni/commons/events/ui/JumpToDatabaseListRequestEvent.java]: Fixed

**Pull request merged**